.github/workflows/makefile.yml

```diff
@@ -0,0 +1,280 @@
+name: Makefile CI
+
+on:
+  push:
+  pull_request:
+env:
+  # BASIC_PACKAGES is used fo the basic tools (make)
+  BASIC_PACKAGES: git build-essential ccache libgc-dev graphviz libunwind-dev libreadline-dev pkg-config
+  # MORE_PACKAGES is used for more tools and libs (make more)
+  MORE_PACKAGES: libgmp-dev libcurl4-openssl-dev libevent-dev openjdk-17-jdk-headless libgles-dev libegl-dev libsdl1.2-dev libsdl2-dev libsdl2-mixer-dev libsdl2-gfx-dev libsdl2-image-dev libsdl2-net-dev libsdl2-ttf-dev libgtk-3-dev inkscape postgresql libxdg-basedir-dev gettext libsqlite3-dev libpq-dev libmongoc-dev openmpi-bin libopenmpi-dev
+  JAVA_HOME: /usr/lib/jvm/java-17-openjdk-amd64/
+  JNI_LIB_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+  LD_LIBRARY_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES
+    - run: make
+    - uses: actions/upload-artifact@v4
+      with:
+        path: bin
+        name: bin
+
+  sanity-check:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - run: misc/jenkins/ensuregit.sh
+    - run: misc/jenkins/checkwhitespaces.sh
+      if: success() || failure()
+    - run: misc/jenkins/checklicense.sh
+      if: success() || failure()
+    - run: misc/jenkins/checksignedoffby.sh
+      if: success() || failure()
+
+  tests-some:
+    runs-on: ubuntu-24.04
+    needs: [build]
+    container: "ubuntu:24.04"
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+      mongo:
+        image: mongo
+      neo4j:
+        image: 'neo4j:2.3'
+        env:
+          NEO4J_AUTH: none
+    strategy:
+      fail-fast: false
+      matrix:
+        engine: [nitcg, nitcs, nitcsg, nitce, niti]
+    steps:
+    - name: Install dependencies
+      run: apt-get update && apt-get install -y $BASIC_PACKAGES $MORE_PACKAGES
+    - uses: actions/checkout@v4
+    - run: misc/jenkins/ensuregit.sh
+    - uses: actions/download-artifact@v4
+    - run: chmod +x bin/*
+    - run: make src/version.nit
+    - run: cd tests && ./search_tests_git.sh origin/master HEAD > list
+    - run: cd tests && ./tests.sh --engine ${{ matrix.engine }} `cat list`
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: 'tests/*.xml'
+        name: tests-some-${{ matrix.engine }}
+    - uses: actions/upload-artifact@v4
+      if: success() || failure()
+      with:
+        path: 'tests/out'
+        name: tests-some-${{ matrix.engine }}-res
+    - uses: EnricoMi/publish-unit-test-result-action@v2
```

Comment:
There was a weird message in the logs, since this is not a fork. Investigate?

```
2024-06-26 15:53:42 +0000 - publish -  INFO - This action is running on a pull_request event for a fork repository. Pull request comments and check runs cannot be created, so disabling these features. To fully run the action on fork repository pull requests, see https://github.com/EnricoMi/publish-unit-test-result-action/blob/v2.16.1/README.md#support-fork-repositories-and-dependabot-branches
```

Reply:
It is a fork, as the PR is from privat/mynit to nitlang/nit, nevertheless previous experiments with the EnricoMi/publish-unit-test-result-action was done without a PR/fork, and now I see thant to you that the results of our numerous tests are not publlished.

The complete setup in the README is complex and maybe github action is not yet production ready :(
So I will inverstigate more.